includes/compat/class-wc-stripe-sepa-subs-compat.php

```diff
@@ -85,17 +85,21 @@ public function is_subs_change_payment() {
 	public function display_update_subs_payment_checkout() {
 		if (
 			apply_filters( 'wc_stripe_display_update_subs_payment_method_card_checkbox', true ) &&
-			wcs_user_has_subscription( get_current_user_id(), '', 'active' ) &&
+			wcs_user_has_subscription( get_current_user_id(), '', apply_filters( 'wc_stripe_display_update_subs_payment_method_card_statuses', array( 'active' ) ) ) &&
```

Comment:
This filter is pointless by itself, because only `'active'` subscriptions' payment methods will be updated ([see here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/740/files#diff-0e15c5ae3c4b28bb406663efdd18e1d2R115)).

So we should either remove this filter, or apply it too in the `handle_add_payment_method_success` functions.

@roykho Is there any reason why, instead of searching for `active` subscriptions, we could search for `active` **or** `on-hold`? That should cover the basic use case for this change I think (allowing the user to add a new credit card when the previous one failed in an existing subscription).

Reply:
@DanReyLop by itself, this filter is indeed not necessary for the majority of users.

We need this filter so we can resolve #728 with custom code and have subscriptions of any status (not just active) use the new payment details, if a customer updates their card.